### PR TITLE
Adding `BenchPOTS` from PyPI 

### PR DESCRIPTION
**Pull request by WenjieDu**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by WenjieDu** [comment]
For Win-64, the PyTorch dependency is missing, but I still would like to remain this package noarch:

``` console
WARNING: failed to get package records, retrying.  exception was: Unsatisfiable dependencies for platform win-64: {MatchSpec("pytorch[version='>=1.10.0']")}
Encountered problems while solving:
  - nothing provides requested pytorch >=1.10.0
```

This PR is ready, @conda-forge-admin, please ping conda-forge/help-python to help review and merge.

File: recipes/benchpots/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 12726b7db046ba0f3ca04a4be8b3a3075608e55fb666a85791d97d91d59a43d6
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (ocefpaf) [inline recipes/benchpots/meta.yaml:14]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

File: recipes/benchpots/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - python >=3.8
+    - pip
+    - h5py
+    - numpy
+    - pandas
+    - pyarrow
+    - scikit-learn
+    - pytorch >=1.10.0
+    - tsdb >=0.4
+    - pygrinder >=0.6
```
Comment (ocefpaf) [inline recipes/benchpots/meta.yaml:28]: These are all run dependencies, no? Unless the package imports itself at build time, which you should report upstream b/c it is a bug, you don't need these here not in the test section. You do need them in the run section though.

```suggestion
```

File: recipes/benchpots/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  run:
+    - python >=3.8
```
Comment (ocefpaf) [inline recipes/benchpots/meta.yaml:30]: These are the dependencies listed in the package metadata. See https://github.com/WenjieDu/BenchPOTS/blob/v0.1.1/setup.py#L47-L53

Note that they are also listed as extra requirement in the setup.cfg, which is conflicting with the definition fro the setup.py.

```suggestion
    - python >=3.8
    - h5py
    - numpy
    - pandas
    - scikit-learn
    - tsdb >=0.4
    - pygrinder >=0.6
```

File: recipes/benchpots/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  requires:
+    - pip
+    - h5py
+    - numpy
+    - pandas
+    - pyarrow
+    - scikit-learn
+    - pytorch >=1.10.0
+    - tsdb >=0.4
+    - pygrinder >=0.6
```
Comment (ocefpaf) [inline recipes/benchpots/meta.yaml:44]:
```suggestion
```

**Review by ocefpaf — CHANGES_REQUESTED** [review]

**Comment by WenjieDu** [comment]
@ocefpaf Many thanks for your quick response, Filipe. I've accepted and applied all your code suggestions.

**Pull request merged**